src/main/kotlin/ch/bfh/habits/services/HabitService.kt

```
@@ -10,62 +10,28 @@ import org.springframework.stereotype.Service
-                val group = groupService.getGroup(groupId, userId)
-                currentHabit.group = group
-            } else {
-                currentHabit.group = null
-            }
-        }
+    @Transactional fun e(i: Long, u: Long) = h.delete(d(i, u))
 
-        HabitEntityBuilder.applyHabitDtoToEntity(habitDTO, currentHabit)
-        return currentHabit
+    @Transactional fun f(i: Long, d: HabitDTO, u: Long): Habit {
+        if ((d.frequencyValue != null && d.frequency == null) || (d.frequencyValue == null && d.frequency != null)) throw BadRequestException("Frequency and frequencyValue must be set together")
```

detekt (1.23.7)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)